tests/ut/runtime/test_task_submit_dispatch.py

```
@@ -0,0 +1,223 @@
+
+@pytest.fixture(autouse=True)
+def _reset_pipeline_ctx():
+    """Isolate the module-global pipeline ctx / pools between tests."""
+    saved_ctx = dict(test_runner._pipeline_ctx)
+    saved_pool = test_runner._device_pool
+    test_runner._pipeline_ctx.clear()
+    test_runner._executed_device.clear()
+    yield
+    test_runner._pipeline_ctx.clear()
+    test_runner._pipeline_ctx.update(saved_ctx)
+    setattr(test_runner, "_device_pool", saved_pool)
```

Ruff (0.15.18)

[warning] Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)